docs/content/preview/releases/techadvisories/ta-21297.md

```diff
@@ -0,0 +1,97 @@
+---
+title: TA-21297
+headerTitle: Missing writes during Batch Execution in a transaction
+headcontent: 25 March 2024
+type: docs
+showRightNav: true
+cascade:
+  unversioned: true
+menu:
+  preview_releases:
+    identifier: ta-21297
+    #parent: tech-advisories
+    weight: 1
+rightNav:
+  hideH2: true
+type: docs
+---
+
+|          Product           |  Affected Versions  |  Related Issues   | Fixed In |
+| :------------------------- | :------------------ | :---------------- | :------- |
+| {{<product "ysql">}}       | {{<release "All">}} | {{<issue 21297>}} | N/A      |
+
+## Description
+
+Due to a day-0 issue with transaction retries it is possible that, in specific circumstances, a transaction is _partially_ executed and no error is returned. This is an atomicity violation and may result in data loss.
+
+This can happen in the following circumstance:
+
+- You use some form of write "batch statement" (for example, multiple write statements into the same message) specifically as the first message/statement in the transaction; and a transaction conflict occurs for any but the first internal statement in that batch statement.
+
+## Mitigation
+
+The workaround is to add a read statement (for example, `SELECT 1`) as the first statement in each transaction.
+
+You can also disable the transaction retries according to the installed release as follows:
+
+- Versions prior to {{<release "2.18.5.1, 2.18.6.0, 2.20.1.1,2.20.2.0, 2.21.0.0">}}.
+
+    Set the following flags:
+
+    ```sql
+    ysql_max_read_restart_attempts = 0
+    ysql_max_write_restart_attempts = 0
+    ```
+
+- Versions starting from {{<release "2.18.5.1">}}, {{<release "2.18.6.0">}}, {{<release "2.20.1.1">}}, {{<release "2.20.2.0">}}, and {{<release "2.21.0.0">}}.
```

Comment:
```suggestion
- Versions starting from {{<release "2.18.5.1, 2.18.6.0, 2.20.1.1, 2.20.2.0, and 2.21.0.0">}}
```

Reply:
@premkumr - is this correct as well? To include "and" inside or should it be:
- Versions starting from {{<release "2.18.5.1, 2.18.6.0, 2.20.1.1, 2.20.2.0">}}, and {{<release "2.21.0.0">}}

Reply:
It does not support and. For now , we can use it the way you way suggested. We can discuss this later.